src/actions/classifier.js

```diff
@@ -35,3 +51,139 @@ export function fetchWorkflow(workflowID) {
+      if (getState().user.isGuestUser){
+        return resolve()
+      }
+
+      dispatch(getAuthUser()).then((userResource)=> {
+        userResource.get('project_preferences', {project_id: projectID}).then (([projectPreferences]) => {
+          //Before being able to classify on a project, the user needs to have their preference created if it doesn't exist
+          if (projectPreferences) {
+            return resolve()
+          }
+
+          const projectPreference = {
+            links: { project: projectID },
+            preferences: {}
+          }
+
+          apiClient.type('project_preferences').create(projectPreference).save().then(() => {
+            dispatch(setState(`user.projects.${projectID}`, {
+                name: project.display_name,
+                slug: project.slug,
+                activity_count: 0,
+                sort_order: '',
+                tutorials_completed_at: {}
+              }
+            ))
+            return resolve()
+          }).catch(() => {
+            return reject()
+          })
+        })
+      })
+    })
+  }
+}
+
+export function setNeedsTutorial() {
+  return (dispatch, getState) => {
+    return new Promise ((resolve) => {
+      const workflowID = getState().classifier.currentWorkflowID
+      if (!getState().classifier.tutorial[workflowID]) {
+        return resolve()
+      }
+
+      const projectID = getState().classifier.workflow[workflowID].links.project
+      const tutorialID = getState().classifier.tutorial[workflowID].id
+      let needsTutorial = getState().classifier.needsTutorial[workflowID] !== undefined ? getState().classifier.needsTutorial[workflowID] : true
+
+      if ((!getState().user.isGuestUser) && (getState().user.projects[projectID])) {
+        needsTutorial = !getState().user.projects[projectID]['tutorials_completed_at'][tutorialID]
+      }
+
+      dispatch(setState(`classifier.needsTutorial.${workflowID}`, needsTutorial))
+      return resolve()
+    })
+  }
+}
+
+export function setTutorialCompleted() {
+  return (dispatch, getState) => {
+    const workflowID = getState().classifier.currentWorkflowID
+    dispatch(setState(`classifier.needsTutorial.${workflowID}`, false))
+
+    if (getState().user.isGuestUser) {
+      return
+    }
+    const now = new Date().toISOString()
+    const tutorialID = getState().classifier.tutorial[workflowID].id
+    const projectID = getState().classifier.workflow[workflowID].links.project
+
+    dispatch(getAuthUser()).then((userResourse) => {
+      userResourse.get('project_preferences', {project_id: projectID}).then (([projectPreferences]) => {
+        if (!projectPreferences.preferences.tutorials_completed_at) {
+          projectPreferences.preferences.tutorials_completed_at = {}
+        }
+        const completed = {
+          [tutorialID]: now
+        }
+        projectPreferences.update({
+          preferences: {
+            tutorials_completed_at: completed
```

Comment:
They're tracked by tutorial id on PFE since tutorials belong to workflows: https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/components/tutorial.cjsx#L114-L115